src/iceberg/type_fwd.h

```diff
@@ -81,4 +81,36 @@ class TimestampTzType;
 class Type;
 class UuidType;
 
+/// \brief Error types for iceberg.
+/// TODO: add more and sort them based on some rules.
+enum class ErrorKind {
```

Comment:
Should we create a new file for this ErrorKind?

Reply:
Yes, I was also thinking of adding a detail error message like:

```
struct Error {
  ErrorKind kind;
  std::string message;
};
```

Reply:
This seems same as `Status`?

Reply:
Status can be OK but here we don't want to include that.